### PR TITLE
Debugging

### PR DESCRIPTION
**Pull request by orlichter1**
## Status
Ready

## Description
Moving things from internal conetent

### CODE REVIEW

**Comment by orlichter1** [comment]
@anara123 Added the links in the readme

**Comment by kirbles19** [comment]
@orlichter1 Done.

**Comment by anara123** [comment]
@orlichter1 the build fails. I think it is because you have directory with space in it. change that directory name

**Comment by orlichter1** [comment]
@anara123 Hey I added "[Design](https://github.com/demisto/content/tree/Internal-public/docs/integration_design)" (and renamed it as "IntegrationDesign") and "[Playbook Conventions](https://github.com/demisto/content/tree/Internal-public/docs/playbook_conventions)". Could you have a look and see that it fits what you wanted?
p.s. I already updated the links to look in master branch so they might not work in this one.

File: docs/integration_design/README.MD [hunk 1]
```diff
@@ -0,0 +1,104 @@
+# Integration Design
+## Intro
+In this tutorial we'll talk about how to get from an initial need or idea for an integration you want to implement in Demisto, to a design document that'll help you get organized and understand exactly what you need to implement.
+
+
+## Why do I need a design document in the first place?
+You might be thinking "Why do I need to write this document? Why not just start writing?" 
+Well, we beleive that writing such a document before starting any code writing will help you better understand your use of this integration, understand what exactly you want to do and what you want to get out of each command or fetch.
```
Comment (anara123) [inline docs/integration_design/README.MD]:
```suggestion
Well, we believe that writing such a document before starting any code writing will help you better understand your use of this integration, understand what exactly you want to do and what you want to get out of each command or fetch.
```

File: docs/integration_design/README.MD [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+Write down the use cases for your integration, how you would use it and how you think outher people will use it.
+```Markdown
+# Use Cases:
+- Create, get, edit, close a ticket/issue, add + view comments.
+- Assign a ticket/issue to a specified user.
+- List all tickets, filter by name, date, assignee
+```
+## Parameters
+In this section write what are the parameters you need in order to create an instance of the product and connect to it. These parameters will be accessible and used in all the integration. 
+We require 2 fields as part of the parameters:
```
Comment (anara123) [inline docs/integration_design/README.MD]: mention common parameters that we see in integrations, like `url`, `apikey` / `apitoken`/ `credentials (username, password)`, if the integrations fetches incidents then `query` which will query events/incidents/tickets

File: docs/integration_design/README.MD [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+
+When implementing the integration have a look at existing integrations to see how we implement this.
+If your integration implements `Fetches incidents` also add a boolean field called `isFetch`
+## Commands, Arguments & Outputs
+Now let's write down exactly what commands you want to implement. Each command has 3 parts:
+1. Command name
+2. Command arguments
+3. Command outputs
+#### Command Name
+We have a standard for a command's name: `productName-command`
```
Comment (anara123) [inline docs/integration_design/README.MD]:
```suggestion
We have a standard for a command's name: `brand-action`
```

File: docs/integration_design/README.MD [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+For example `intsights-get-alert-activities`
+If your commands checks the maliciousness of an IP/Domain/URL/File name your command simply `ip`\ `domain`\ `url`\ `file`. If this is the case make sure your outputs (which we'll get to in a second) match the [context standards](https://github.com/demisto/content/tree/master/docs/context_standards).
+
+#### Command argument
+These are the arguments for the command for example `!intsights-get-alert-activities alert-id=`
```
Comment (anara123) [inline docs/integration_design/README.MD]:
```suggestion
These are the arguments for the command for example `!intsights-get-alert-activities alert-id=`, `!ip ip="8.8.8.8"`
```

File: docs/integration_design/README.MD [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+
+#### Command Outputs
+There are 3 type of outputs from the command:
+1. Human Readable output - This is what will be printed to the war room. It should be in Markdown form.
+2. Raw Json - A JSON of the raw response that was returned from the API. 
+3. Context Output - The outputs that will be in the context. 
+
+The Context Outupt will be a JSON in the following format `{Product.TypeofOuput.Info}`. For more info about this have a look [here](https://github.com/demisto/content/tree/master/docs/context_and_ouputs)
+
+### Command Deisgn
```
Comment (anara123) [inline docs/integration_design/README.MD]:
```suggestion
### Command Design
```

File: docs/integration_design/README.MD (hunk 4 above)
Comment (anara123) [inline docs/integration_design/README.MD]: argument names must be snake case. 
good: `alert-id`.
bad: `alertId`

File: docs/integration_design/README.MD [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+The final result for each command should look something like this:
+
+**Command Name**: cisco-asa-list-rules
+
+###### Arguments
+argument | type | description|mandatory
+---|---|---|---
+InterfaceName|String|the name of the interface|no
```
Comment (anara123) [inline docs/integration_design/README.MD]: InterfaceName is not according to our standard.

File: docs/integration_design/README.MD [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+* CiscoASA.Rule.Active
+* CiscoASA.Rule.Remarks
+* CiscoASA.Rule.Position
+* CiscoASA.Rule.ID
+###### Human Readable Output
+**All Rules for interface {} of type{}**:
+
+RuleID|InterfaceName|SourceIP|DestIP|Position|
+---|---|---|---|---|
+...|...|...|...|...|
```
Comment (anara123) [inline docs/integration_design/README.MD]: present some data here

**Comment by orlichter1** [comment]
@anara123 Added the Command examples and changed your requests for Integration Design :)

**Comment by orlichter1** [comment]
@amshamah419 could you please go over [this](https://github.com/demisto/content/edit/Internal-public/docs/integration_design/README.MD?pr=%2Fdemisto%2Fcontent%2Fpull%2F4608)

**Comment by amshamah419** [comment]
Not quite done yet, will finish tomorrow.

**Comment by amshamah419** [comment]
Looks good @anara123 @orlichter1

**Pull request merged**